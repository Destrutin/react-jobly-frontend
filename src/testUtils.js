import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  password: "omgsecret",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
