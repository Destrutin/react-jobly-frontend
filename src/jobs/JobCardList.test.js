import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {UserProvider} from "../testUtils";
import JobCardList from './JobCardList';

const jobs = [
    {id: 1, title: "test", salary: "1", equity: "1", companyName: "test"}
]

test('renders job card list without crashing', () => {
    render(
        <MemoryRouter>
            <UserProvider>
            <JobCardList jobs={jobs}/>
            </UserProvider>
        </MemoryRouter>
    )
})

test('matches snapshot', () => {
    const {asFragment} = render(
        <MemoryRouter>
            <UserProvider>
            <JobCardList jobs={jobs}/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})