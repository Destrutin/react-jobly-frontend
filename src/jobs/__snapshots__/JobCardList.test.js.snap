// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCardList"
  >
    <div
      class="JobCard-card"
    >
      <div
        class="JobCard-body"
      >
        <h4>
          test
        </h4>
        <p>
          test
        </p>
        <p>
          Salary: 1
        </p>
        <p>
          Equity: 1
        </p>
        <button
          class="apply-button"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
